addons/mail/static/src/thread/message.js

```diff
@@ -4,23 +4,45 @@ import { useMessaging } from "../messaging_hook";
 import { RelativeTime } from "./relative_time";
 import { Component } from "@odoo/owl";
 import { useService } from "@web/core/utils/hooks";
+import { MessageDeleteDialog } from '@mail/thread/message_delete_dialog';
 
 export class Message extends Component {
     static template = "mail.message";
     static components = { RelativeTime };
-    static props = ["message", "squashed?"];
+    static props = ["disableActions?", "message", "squashed?"];
 
     setup() {
         this.messaging = useMessaging();
         this.action = useService("action");
+        this.user = useService("user");
         this.message = this.props.message;
         this.author = this.messaging.partners[this.message.authorId];
     }
 
+    get canBeDeleted() {
+        if (this.props.disableActions) {
+            return false;
+        }
+        if (!this.user.isAdmin && this.message.authorId !== this.messaging.user.partnerId) {
```

Comment:
`this.messaging.user.partnerId` can just be `this.user.partnerId`.
Only access `this.messaging.user` for messaging-specific data on current user. If available in `user_service`, rely on data in this latter service instead.